src/drivers/ncbbio/ncbbio_log_put.c

```diff
@@ -351,7 +351,12 @@ int ncbbio_log_put_varn(NC_bb            *ncbbp,
     * Include metadata entry header and variable size additional data
     * (start, count, stride)
     */
-    esize = sizeof(NC_bb_metadataentry) + ndims * 2 * SIZEOF_MPI_OFFSET * num;
+    if (counts != NULL){
+        esize = sizeof(NC_bb_metadataentry) + ndims * 2 * SIZEOF_MPI_OFFSET * num;
+    }
+    else{
+        esize = sizeof(NC_bb_metadataentry) + ndims * SIZEOF_MPI_OFFSET * num;
```

Comment:
When counts == NULL, we still need to allocate space for counts and fill it will all 1s.

Reply:
Since the driver are required to take NULL counts, ncmpio driver should have ability to handle it as well. We can omit counts in the metadata log entirely if counts is NULL. When flushing, we check the metadata entry size to tell if countss is included. If not, we pass NULL to ncmpio.